packages/create-cloudflare/src/wrangler/__tests__/accounts.test.ts

```diff
@@ -0,0 +1,169 @@
+import { mockPackageManager, mockSpinner } from "helpers/__tests__/mocks";
+import { runCommand } from "helpers/command";
+import { beforeEach, describe, expect, test, vi } from "vitest";
+import { isLoggedIn, listAccounts, wranglerLogin } from "../accounts";
+
+const loggedInWhoamiOutput = `
+-------------------------------------------------------
+Getting User settings...
+👋 You are logged in with an OAuth Token, associated with the email person@mail.com!
+┌──────────────┬──────────────────────────────────┐
+│ Account Name │ Account ID                       │
+├──────────────┼──────────────────────────────────┤
+│ testacct     │ g8s9dl23jv90xa0xxxx990ds09xxxxda │
+└──────────────┴──────────────────────────────────┘
+`;
+
+const loggedOutWhoamiOutput = `
+-------------------
+Getting User settings...
+You are not authenticated. Please run \`wrangler login\`.
+`;
+
+const loginDeniedOutput = `
+✘ [ERROR] Error: Consent denied. You must grant consent to Wrangler in order to login.
+`;
+
+const loginSuccessOutput = `
+Successfully logged in.
+`;
+
+vi.mock("helpers/command");
+vi.mock("which-pm-runs");
+vi.mock("@cloudflare/cli/interactive");
+
+describe("wrangler account helpers", () => {
+	let spinner: ReturnType<typeof mockSpinner>;
+
+	beforeEach(() => {
+		spinner = mockSpinner();
+	});
+
+	describe("wranglerLogin", async () => {
+		test("logged in", async () => {
+			mockPackageManager("npm");
+			const mock = vi
+				.mocked(runCommand)
+				.mockReturnValueOnce(Promise.resolve(loggedInWhoamiOutput));
+
+			const loggedIn = await wranglerLogin();
+
+			expect(loggedIn).toBe(true);
+			expect(mock).toHaveBeenCalledWith(
+				["npx", "wrangler", "whoami"],
+				expect.anything()
```

Comment:
This is because each call to `runCommand` will also include an options object with things like `env` and `silent` which we don't care about so much. In many cases we just want to validate whether a command with the correct arguments were called.

Reply:
ah! duh yeah :sweat_smile:

sorry my bad, I misread the code and thought this was for the return value :sweat: :bow: